File: springboot-park/src/main/java/com/example/demo/service/VehicleService.java
package com.example.demo.service;

import java.util.List;

import com.example.demo.dto.VehicleDto;

public interface VehicleService {
	
	VehicleDto createVehicle(VehicleDto vehicleDto);
	
	VehicleDto getVehicleByLot(Long lot);
	
	List<VehicleDto> getAllVehicle();
	
	VehicleDto updateVehicle(Long lot, VehicleDto updateVehicle);

	void deleteVehicle(Long lot);
}

File: frotend/src/app/vehicle-service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Vehicle } from './vehicle';



@Injectable({
  providedIn: 'root',
})
export class VehicleService {
  private baseURL = "http://localhost:8080/smartpark/vehicle"

  constructor(private httpClient: HttpClient) {}

  getVehicleList(): Observable<Vehicle[]>{
    return this.httpClient.get<Vehicle[]>(`${this.baseURL}`);     
  }

  createVehicle(vehicle: Vehicle): Observable<Object>{
    return this.httpClient.post(`${this.baseURL}`, vehicle);
  }

  getVehicleByLot(lot: number): Observable<Vehicle>{
     return this.httpClient.get<Vehicle>(`${this.baseURL}/${lot}`);
  }

   updateVehicle(lot: number, vehicle: Vehicle): Observable<Object>{
    return this.httpClient.put(`${this.baseURL}/${lot}`, vehicle);
  }

  deleteVehicle(lot: number): Observable<Object>{
    return this.httpClient.delete(`${this.baseURL}/${lot}`);
  }

}

